Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -475,143 +469,201 @@ namespace OpenInfraPlatform {
+			*/
 			// Function 6: Convert IfcParametrizedProfileDef
-			void convertIfcParameterizedProfileDef(const std::shared_ptr<typename IfcEntityTypesT::IfcParameterizedProfileDef>& profileDef,
+			void convertIfcParameterizedProfileDef(const EXPRESSReference<typename IfcEntityTypesT::IfcParameterizedProfileDef>& profileDef,
 				std::vector<std::vector<carve::geom::vector<2>>>& paths) const
 			{
 
-				// *************************************************************************************************************************************************************//
-				// IfcParameterizedProfileDef	(http://www.buildingsmart-tech.org/ifc/IFC4x1/final/html/schema/ifcprofileresource/lexical/ifcparameterizedprofiledef.htm)		//
-				// ABSTRACT SUPERTYPE OF IfcCShapeProfileDef, IfcCircleProfileDef, IfcEllipseProfileDef, IfcIShapeProfileDef, IfcLShapeProfileDef,								//
-				// IfcRectangleProfileDef, IfcTShapeProfileDef, IfcTrapeziumProfileDef, IfcUShapeProfileDef, IfcZShapeProfileDef												//
-				// *************************************************************************************************************************************************************//
+				// *************************************************************************************************************************************************************
+				// ENTITY IfcParameterizedProfileDef	
+				// https://standards.buildingsmart.org/IFC/DEV/IFC4_2/FINAL/HTML/schema/ifcprofileresource/lexical/ifcparameterizedprofiledef.htm		
+				// ABSTRACT SUPERTYPE OF (IfcCShapeProfileDef, IfcCircleProfileDef, IfcEllipseProfileDef, IfcIShapeProfileDef, IfcLShapeProfileDef,								
+				// IfcRectangleProfileDef, IfcTShapeProfileDef, IfcTrapeziumProfileDef, IfcUShapeProfileDef, IfcZShapeProfileDef)	
+				// SUBTYPE OF (IfcProfileDef);
+				// END_ENTITY
+				// *************************************************************************************************************************************************************
 #ifdef _DEBUG
 				BLUE_LOG(trace) << "Processing IfcParameterizedProfileDef #" << profileDef->getId();
 #endif
-				double length_factor = UnitConvert()->getLengthInMeterFactor();
-				double angle_factor = UnitConvert()->getAngleInRadianFactor();
-				std::vector<carve::geom::vector<2>> outer_loop;
 
 				// (1/10) IfcRectangleProfileDef SUBTYPE OF IfcParametrizedProfileDef
-				std::shared_ptr<typename IfcEntityTypesT::IfcRectangleProfileDef> rectangle_profile = std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcRectangleProfileDef>(profileDef);
-				if (rectangle_profile) {
-					convertIfcRectangleProfileDef(EXPRESSReference<typename IfcEntityTypesT::IfcRectangleProfileDef>(rectangle_profile), paths, outer_loop);
+				if (profileDef.isOfType<typename IfcEntityTypesT::IfcRectangleProfileDef>()) {
+					convertIfcRectangleProfileDef(profileDef.as<typename IfcEntityTypesT::IfcRectangleProfileDef>(), paths);

```

Comment:
Since there is no `return` in these `if` clauses, the `throw` at the end of function will always happen. Include returns.

Reply:
Resolved: 8838897